### DISCUSSION TITLE
Proposal: Publish all NPM packages with the same version number

### DISCUSSION THREAD

**Discussion by johnhooks** [WordPress/gutenberg#49140] [Ideas]
With the current versioning system, it is difficult to understand what packages are compatible with each other. When initializing a new project it's simple to just install all the latest versions, but then as time goes by it becomes harder to distinguish what versions are compatible or even what version of Gutenberg they are from. To figure it out requires diving into the `package.json` files and/or doing unnecessary research.

I propose all packages be published using the same version numbering as Gutenberg.

If packages need to be published between version releases they could use the alpha or beta tags, such as `1.1.1-alpha.1` or `1.1.1-beta.1`.

Edit:

> We already have the RC and beta's outside of the version (like 15.3 beta, 15.3 RC1-2-3 etc.)

@tomdevisser I agree, RC and beta postfix versioning are established. I am still getting familiar with this project and didn't know the conventions used, I suggested the others because it is what I've seen used in other projects.

**Comment by ghost** [comment]
I don't think WordPress even uses semantic versioning, I think it's major.major.minor. I don't know how it translates to Gutenberg, but I think it would be nice to transfer to semantic versioning.

I'm not sure about the `1.1.1-alpha.1` syntax, as we already have the RC and beta's outside of the version (like 15.3 beta, 15.3 RC1-2-3 etc.). I do like the idea of thinking about ways to make it more clear what packages are compatible with each other. Doesn't npm update automatically find/mention compatible versions though?

**Reply by johnhooks** [reply]
If there is a established postfix to the version, I agree that should be used. In theory npm update should work, though it would be a nightmare to figure out if it doesn't. I wouldn't want to run that on project without knowing for sure they were all correct, and it's impossible to tell at a glance that they are. What I'm suggesting is identical to how `react` and `react-dom` are versioned. Many other projects follow this same pattern. It makes updates of multiple dependencies reliable and predictable. If all of the `@wordpress/*` dependencies are the same it's easy to tell they should work together. It also makes targeting older versions of Gutenberg easier.

**Comment by johnhooks** [comment]
I just realized we are probably stuck with this system because some packages already have higher version numbers than Gutenberg, `@wordpress/components` is currently at `v23.5.0`.

If publishing all the package with the same version isn't an option. How else can we know what packages should go together? The `package.json` in the repo isn't much help, it uses relative file urls. 

Besides writing a script to crawl all the `package.json` files of a specific release branch, is this inform available anywhere? Similar to the version chart below, comparing Gutenberg releases to WordPress, but a listing of all package versions at time of release?

https://developer.wordpress.org/block-editor/contributors/versions-in-wordpress/

**Comment by dmsnell** [comment]
I believe that this is mostly related to #46693, or at least is close enough that the discussion can all take place there.
